File: TirePressureMonitoring/C/alarm.c
#include "alarm.h"
#include "sensor.h"

struct Alarm_t AlarmData;

void Alarm_Init(void)
{
    Sensor_Init();
    AlarmData.lowPressureTreshold = 17;
    AlarmData.highPressureTreshold = 21;
    AlarmData.alarmOn = false;
}

void Alarm_Check(void)
{
    double psiPressureValue = Sensor_PopNextPressurePsiValue();

    if (psiPressureValue < AlarmData.lowPressureTreshold || AlarmData.highPressureTreshold < psiPressureValue)
    {
        AlarmData.alarmOn = true;
    }
}

bool Alarm_IsAlarmOn(void)
{
    return AlarmData.alarmOn;
}

File: TirePressureMonitoring/Swift/Alarm.swift
import Foundation

public class Alarm {
    enum Thesholds {
        static let lowPressure = 17.0
        static let highPressure = 21.0
    }

    let sensor = Sensor()
    
    var alarmOn = false
    
    public func check() -> Void {
        let psiPressureValue = sensor.popNextPressurePsiValue()
        
        if psiPressureValue < Thesholds.lowPressure ||
            Thesholds.highPressure < psiPressureValue {
            alarmOn = true
        }
    }
    
    public var isAlarmOn: Bool {
        return self.alarmOn
    }
}
